packages/components/src/Inject/Inject.component.js

```diff
@@ -113,17 +113,34 @@ Inject.getAll = function injectGetAll(getComponent, config) {
  * @param {function} getComponent
  * @param {object|string|React Element} data
  */
-Inject.getReactElement = function getReactElement(getComponent, data, CustomInject = Inject) {
+Inject.getReactElement = function getReactElement(
+	getComponent,
+	data,
+	CustomInject = Inject,
+	withKey,
+) {
 	if (Array.isArray(data)) {
-		return data.map(info => getReactElement(getComponent, info, CustomInject));
+		return data.map(info => getReactElement(getComponent, info, CustomInject, true));
 	} else if (data === null) {
 		return data;
 	} else if (typeof data === 'string') {
-		return <CustomInject getComponent={getComponent} component={data} />;
+		return (
+			<CustomInject
+				getComponent={getComponent}
+				component={data}
+				key={withKey && `${data}#default`}
```

Comment:
if `withKey` is false, `key` value will be `undefined` which may happen on multiple components and React will complain about having sibling elements with the same `undefined` `key`, it's maybe better to not define this attritute at all when `withKey` is false